src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkService.java

```
@@ -28,6 +30,34 @@ public class LearningWalkService {
   @Autowired private DateProvider dateProvider;
   @Autowired private LabService labService;
 
+  @Transactional
+  public LearningWalkClassroom saveClassroomDemographics(
+      UUID lwID, Long classroomId, SaveLearningWalkClassroomDemographicsRequest demographics) {
+    LearningWalk learningWalk =
+        learningWalkRepository
+            .findByUuid(lwID)
+            .orElseThrow(() -> new LearningWalkNotFoundException(lwID));
+    List<LearningWalkClassroom> learningWalkClassrooms =
+        learningWalkClassroomsRepository.findByLearningWalkId(learningWalk.getId());
+
+    LearningWalkClassroom classroom =
```

Why not just search by classroom id and throw an exception if not found? 
`
LearningWalkClassroom classroom = learningWalkClassroomRepository.findById(classroomId).orElseThrow(() -> new LearningWalkClassroomNotFoundException(classroomId));`